go/libraries/doltcore/sqle/dprocedures/dolt_branch.go

```diff
@@ -53,7 +53,11 @@ func doDoltBranch(ctx *sql.Context, args []string) (int, error) {
 		return 1, fmt.Errorf("Empty database name.")
 	}
 
-	apr, err := cli.CreateBranchArgParser().Parse(args)
+	// CreateBranchArgParser has the common flags for the command line and the stored procedure.
+	// The stored procedure doesn't support all actions, so we have a shorter description for -r.
```

Comment:
Is there a reason for the limitation? Should the stored procedure be 1-1 with the CLI, especially since you are leveraging it?

Reply:
The stored procedure is only for modifying branches. We use system tables for reading them.

The CLI does both. Therefore, it has additional flags.